pkg/csi/service/common/vsphereutil.go

```diff
@@ -605,10 +605,29 @@ func ExpandVolumeUtil(ctx context.Context, manager *Manager, volumeID string, ca
 		log.Infof("Successfully expanded volume for volumeid %q to new size %d Mb.", volumeID, capacityInMb)
 		return "", nil
 	} else {
-		expansionRequired, err := isExpansionRequired(ctx, volumeID, capacityInMb, manager, useAsyncQueryVolume)
+		var expansionRequired bool
```

Comment:
If customer uses the latest driver with this code change and we have already enabled - `improved-csi-idempotency` feature-gate then when will this new code block execute.

are you planning to backport this change to the older release of the driver where the idempotency feature is disabled but the customer is using the latest vSphere 8.0?

Reply:
This code will not be backported to any prior releases of CSI. It will only be executed for Supervisor cluster which uses the same util function. For vanilla, you are right, this code is not required.